src/go/guestagent/pkg/tracker/apitracker.go

```diff
@@ -90,15 +86,21 @@ func (a *APITracker) Add(containerID string, portMap nat.PortMap) error {
 			// The expose API only supports IPv4
 			ipv4, err := isIPv4(portBinding.HostIP)
 			if !ipv4 || err != nil {
+				log.Errorf("did not receive IPv4 for HostIP: %s", portBinding.HostIP)
 				continue
 			}
 
 			log.Debugf("calling %s API for the following port binding: %+v", exposeAPI, portBinding)
 
+			hostSwitchIP, err := hostnameToIPv4(gateway)
+			if err != nil {
+				log.Errorf("could not determine IPv4 address for hostSwitch: %s", portBinding.HostIP)
+				continue
+			}
 			err = a.expose(
 				&types.ExposeRequest{
 					Local:  ipPortBuilder(a.determineHostIP(portBinding.HostIP), portBinding.HostPort),
-					Remote: ipPortBuilder(hostSwitchIP, portBinding.HostPort),
+					Remote: ipPortBuilder(string(hostSwitchIP), portBinding.HostPort),
```

Comment:
`hostSwitchIP` is `hostnameToIPV4(gateway)`, so that's `192.168.127.1`; that used to be `192.168.127.2`. That seems unexpected? It sounds like it wants to be `eth0IP` instead (which isn't a great name either… `virtualInterfaceIP`?)